app/views/screens/AccountScreen.tsx

```diff
@@ -1,37 +1,117 @@
+import { useToast } from "react-native-toast-notifications";
 
 import Screen from "../components/Screen";
 import { defaultStyles, colors } from "../styles";
 import AuthContext, { AuthContextType } from "../../auth/context";
 import asyncStorage from "../../store/asyncStorage";
 import { USER_OBJECT_KEY } from "../../store/constants";
 import TextInput from "../components/TextInput";
+import { UserDetails } from "../../models";
+import useApi from "../../api/hooks/useApi";
+import user from "../../api/user";
+import { RESPONSE_CODES } from "../../api/responseCodes";
+import ActivityIndicator from "../components/ActivityIndicator";
+import { TOAST_TYPE, showNotificationToast } from "../../utility/toastHelper";
+
+const EMAIL_REGEX =
+  /^(([^<>()[\]\\.,;:\s@\"]+(\.[^<>()[\]\\.,;:\s@\"]+)*)|(\".+\"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+
+const PHONE_NUMBER_REGEX = /^[0-9]+$/;
 
 function AccountScreen(props) {
   const authContext: AuthContextType = useContext(AuthContext);
-  const [name, setName] = useState(authContext.user.name);
-  const [phone, setPhone] = useState(authContext.user.phoneNumber);
-  const [email, setEmail] = useState(authContext.user.email);
-  const [address, setAddress] = useState(authContext.user.address);
+  const { name, username, phoneNumber, email, address } = authContext.user;
+  const getUserApi = useApi(user.getUser);
+  const updateUserApi = useApi(user.updateUser);
+  const toast = useToast();
+
+  const {
+    control,
+    handleSubmit,
+    reset,
+    formState: { isDirty },
+  } = useForm<UserDetails>({
+    defaultValues: {
+      name,
+      phoneNumber,
+      email,
+      address,
+      username,
+    },
+  });
 
   const handleLogout = () => {
     authContext.setUserObject(null);
     asyncStorage.removeData(USER_OBJECT_KEY);
   };
 
+  const handleSave = async (data: UserDetails) => {
+    Keyboard.dismiss();
+    try {
+      const updateUser = await updateUserApi.request(data);
+      if (updateUser?.data?.code === RESPONSE_CODES.SUCCESS) {
+        const user = await getUserApi.request(data.username);
+        if (user?.data?.code === RESPONSE_CODES.SUCCESS) {
+          const { name, username, phoneNumber, email, address } = user.data;
+          reset({
+            name,
+            username,
+            phoneNumber,
+            email,
+            address,
+          });
+          authContext.setUserObject(user.data);
+          showNotificationToast(
+            "User account details saved successfully",
+            TOAST_TYPE.SUCCESS,
+            toast
+          );
+          asyncStorage.storeDataObject(USER_OBJECT_KEY, user.data);
+        } else {
+          console.log("Error when calling v1/getUser api - ", user.data);
+          showNotificationToast(
+            "User account details failed to save. Please try again.",
+            TOAST_TYPE.FAILURE,
+            toast
+          );
```

Comment:
I think you can avoid this service call. 
so if UpdateUserApi succeeds then simply the replace older user object with new user object (basically replacing current value with user entered values).

Reply:
Done! I was thinking about consistency so had written it that way. Removed it now.